### ISSUE TITLE
Incorrect indent errors when breaking out of PHP inside an IF statement

### ISSUE THREAD

**Issue by davidfaber** [squizlabs/PHP_CodeSniffer#1101]
Hi there,
Since Version 2.6.2 I get incorrect intendation-errors for certain cases.

If have a template looking like this:

```
<?php
if (somethingIsTrue()) {
    ?>
    <div>
        <?php
        $someVar = doSomething(
            'foobar',
            true,
            'foo',
            'bar'
        );

        if (somethingElseIsTrue()) {
            doOneThing();
        } else {
            doTheOtherThing();
        }
        ?>
    </div>
    <?php
}
?>
```

I get the following errors using the PSR2 standard:
```
----------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 3 LINES
----------------------------------------------------------------------
 13 | ERROR | [x] Line indented incorrectly; expected 4 spaces, found
    |       |     8
 15 | ERROR | [x] Line indented incorrectly; expected 4 spaces, found
    |       |     8
 17 | ERROR | [x] Line indented incorrectly; expected 4 spaces, found
    |       |     8
----------------------------------------------------------------------
PHPCBF CAN FIX THE 3 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

Which is wrong cause everything is indented correctly.

If I do the following, the errors disappear: (adding ?> <?php after line 11)
```
<?php
if (somethingIsTrue()) {
    ?>
    <div>
        <?php
        $someVar = doSomething(
            'foobar',
            true,
            'foo',
            'bar'
        );
        ?>
        <?php
        if (somethingElseIsTrue()) {
            doOneThing();
        } else {
            doTheOtherThing();
        }
        ?>
    </div>
    <?php
}
?>
```

Also if the first function is not multiline the errors disappear:
```
<?php
if (somethingIsTrue()) {
    ?>
    <div>
        <?php
        $someVar = doSomething('foobar', true, 'foo', 'bar');

        if (somethingElseIsTrue()) {
            doOneThing();
        } else {
            doTheOtherThing();
        }
        ?>
    </div>
    <?php
}
?>
```

I hope this is enough information to tackle the issue!

Thx in advance!

**Comment by gsherwood** [comment]
Debug output showing where the error is:
```
$ phpcs temp.php --standard=PSR2 --sniffs=Generic.Whitespace.ScopeIndent --runtime-set scope_indent_debug true

Start with token 0 on line 1 with indent 0
Open scope (T_IF) on line 2
    => indent set to 4 by token 9 (T_OPEN_CURLY_BRACKET)
Close PHP tag found on line 3
    => indent set to 4 by token 12 (T_CLOSE_TAG)
Open PHP tag found on line 5
    => indent set to 8 by token 15 (T_OPEN_TAG)

--- error is here ---
Closing parenthesis found on line 11
    * token is inside condition 1 (T_IF) on line 2 *
    * using condition *
    => checking indent of 4; main indent set to 4 by token 1 (T_IF)
--- error is here ---

[Line 13] Line indented incorrectly; expected 4 spaces, found 8
    => Add adjustment of -4 for token 45 (T_IF) on line 13
Open scope (T_IF) on line 13
    => indent set to 8 by token 53 (T_OPEN_CURLY_BRACKET)
Indent adjusted to 8 for T_STRING on line 14
    => Add adjustment of -4 for token 56 (T_STRING) on line 14
Close scope (T_IF) on line 15
    => indent set to 4 by token 62 (T_CLOSE_CURLY_BRACKET)
[Line 15] Line indented incorrectly; expected 4 spaces, found 8
    => Add adjustment of -4 for token 62 (T_CLOSE_CURLY_BRACKET) on line 15
Open scope (T_ELSE) on line 15
    => indent set to 8 by token 66 (T_OPEN_CURLY_BRACKET)
Indent adjusted to 8 for T_STRING on line 16
    => Add adjustment of -4 for token 69 (T_STRING) on line 16
Close scope (T_ELSE) on line 17
    => indent set to 4 by token 75 (T_CLOSE_CURLY_BRACKET)
[Line 17] Line indented incorrectly; expected 4 spaces, found 8
    => Add adjustment of -4 for token 75 (T_CLOSE_CURLY_BRACKET) on line 17
Close PHP tag found on line 18
    => indent set to 8 by token 78 (T_CLOSE_TAG)
Open PHP tag found on line 20
    => indent set to 4 by token 81 (T_OPEN_TAG)
Close scope (T_IF) on line 21
    => indent set to 0 by token 82 (T_CLOSE_CURLY_BRACKET)
Close PHP tag found on line 22
    => indent set to 0 by token 84 (T_CLOSE_TAG)

FILE: temp.php
------------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 3 LINES
------------------------------------------------------------------------
 13 | ERROR | [x] Line indented incorrectly; expected 4 spaces, found 8
 15 | ERROR | [x] Line indented incorrectly; expected 4 spaces, found 8
 17 | ERROR | [x] Line indented incorrectly; expected 4 spaces, found 8
------------------------------------------------------------------------
PHPCBF CAN FIX THE 3 MARKED SNIFF VIOLATIONS AUTOMATICALLY
------------------------------------------------------------------------

Time: 9ms; Memory: 2Mb
```

**Comment by gsherwood** [comment]
Thanks for the report. I've fixed this case, but please let me know if you find others.

**Comment by davidfaber** [comment]
Thanks mate! That was a really quick fix...

If I find something else I'll let you know!

**Comment by easdue** [comment]
This is happening again in version 3.2.3

**Comment by gsherwood** [comment]
> This is happening again in version 3.2.3

The unit tests for this are all still passing, so you'll need to give me a bit more information about what is failing for you. It would be great if you could open a new issue with some sample code that is failing.
